### PR TITLE
:seedling: Allow ESP protocol to be set in IngressRules

### PR DESCRIPTION
**Pull request by vincepri**
**What type of PR is this?**

**What this PR does / why we need it**:
/kind feature

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [X] squashed commits
- [X] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by JoelSpeed** [comment]
This is required for IPSEC in OVN Kubernetes. I've tested this and it works as expected.
/lgtm

**Comment by Skarlso** [comment]
Hi.

Please include an issue that has been triaged or talked about in any way.

>This is required for IPSEC in OVN Kubernetes. I've tested this and it works as expected.

Can you include any logs or screenshots of the port number being correctly configured?

Thanks.

**Comment by JoelSpeed** [comment]
This was configured in the CNI ingress rules for my AWSCluster:
Screenshot 2023-10-09 at 12 27 59
This is from the status of the AWS Cluster:
Screenshot 2023-10-09 at 12 27 38
This is from the AWS console showing the rules in the security group:
Screenshot 2023-10-09 at 12 28 56

**Comment by Skarlso** [comment]
Thanks! :)

**Review by Ankitasw — COMMENTED** [review]
Can we also add this in one of the existing E2E test to see if it works?

**Comment by vincepri** [comment]
@Ankitasw done

File: test/e2e/data/infrastructure-aws/withoutclusterclass/e2e_test_templates/cluster-template-internal-elb.yaml [hunk 1]
```diff
@@ -28,6 +28,12 @@ spec:
   controlPlaneLoadBalancer:
     scheme: internal
   network:
+    cni:
+      cniIngressRules:
```
Comment (JoelSpeed) [inline test/e2e/data/infrastructure-aws/withoutclusterclass/e2e_test_templates/cluster-template-internal-elb.yaml:32]: IIRC there is a default set of rules that are used when this list is empty, that is not used at all when the list has a single item, you may need to include those default rules to allow the default CNI to continue to function

**Comment by JoelSpeed** [comment]
E2E looking good now
/lgtm

**Pull request merged**